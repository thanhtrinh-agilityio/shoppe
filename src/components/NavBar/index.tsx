import { memo } from 'react'
import { Box, BoxProps } from '@chakra-ui/react'
import Link from 'next/link'

type NavType = {
  value: number
  name: string
  href: string
}

interface NavBarProps extends BoxProps {
  navList: NavType[]
}

const NavBar = ({ navList, ...rest }: NavBarProps) => {
  return (
    <Box
      as="nav"
      textTransform="uppercase"
      display="flex"
      alignItems="center"
      fontSize="16px"
      gap="41px"
      color="secondary"
      {...rest}
    >
      {navList.map((obj) => (
        <Link href={obj.href} key={obj.value}>
          {obj.name}
        </Link>
      ))}
    </Box>
  )
}

export default memo(NavBar)
