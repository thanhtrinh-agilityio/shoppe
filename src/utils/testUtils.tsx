import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
