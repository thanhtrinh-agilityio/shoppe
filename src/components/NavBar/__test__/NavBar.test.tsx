import { render } from '@utils/testUtils'

// Constants
import { NAV_LIST } from '@constants/index'

// Components
import NavBar from '@components/NavBar'

describe('NavBar component', () => {
  it('Renders NavBar correctly', () => {
    const { container } = render(<NavBar navList={NAV_LIST} />)
    expect(container).toMatchSnapshot()
  })
})
