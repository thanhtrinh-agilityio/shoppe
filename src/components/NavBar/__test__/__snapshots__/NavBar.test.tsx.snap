// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component Renders NavBar correctly 1`] = `
<div>
  <nav
    class="css-1aaz8j"
  >
    <a
      href="/contact"
    >
      Contact
    </a>
    <a
      href="/terms-of-services"
    >
      Terms of services
    </a>
    <a
      href="/shipping-and-return"
    >
      Shipping and Return
    </a>
  </nav>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
